app/controllers/api/metrics_controller.rb

```diff
@@ -0,0 +1,69 @@
+class Api::MetricsController < Api::BaseController
+  METRIC_KEYS = %i[
+    host
+    ruby_version
+    bundler_version
+    rubygems_version
+    ruby_engine
+    command
+    git_version
+    rvm_version
+    rbenv_version
+    chruby_version
+  ].freeze
+
+  def create
+    return unless params[:_json]
+
+    validate_data
+    head :ok && return if known_id?(params[:_json].last[:request_id])
+
+    params[:_json].each do |hash|
+      METRIC_KEYS.each do |metric|
+        StatsD.increment("#{metric}.#{hash[metric]}") if hash[metric]
+      end
+      split_increment("options", hash[:options]) if hash[:options]
+      split_increment("ci", hash[:ci]) if hash[:ci]
+    end
+  end
+
+  private
+
+  def split_increment(type, comma_string)
+    comma_string.split(",").each do |val|
+      StatsD.increment("#{type}.#{val}") if val.length < 20
+    end
+  end
+
+  def validate_data
```

Comment:
Okay, I did say people may send garbage data. Sorry I was not clear. 
My intention was that `create` should not throw exceptions. `_json` key not being present was an obvious one, for which using `params.require(:_json)` would be nicer. 
Another is `known_id?(params[:_json].last[:request_id])`. `params[:_json].last` could return nil and `known_id? nil` will also blow up. Please add checks for that.
As far as validation of values using regex goes.. I think we should remove it. Its fine if we store some garbage data, it should be rare and ignorable.

Reply:
Couldn't evil people just spam our database with garbage and mess it up?
Since its all open source, cant someone just find the domain and send thousands of requests to `/api/metrics` with, say, `bundler_version = "drfgredgedrgreger"`?

Reply:
Yes, they could but my guess is that it would be rare. The issue with regex check is that we may end up ignoring valid data.
Attackers are generally interested in the app crashes (exceptions) than messing up data.
If you like we can wait for more opinion on this before removing it :small_airplane:

Reply:
The values we currently store have low cardinality so I think it would be possible to write the regex so that it doesn't ignore valid data.
For example, I've went over all possible Bundler versions to see that they aren't ignored.

I'll let you decide, but I can research farther into possible values and change the regex's to be more permissive if you wish.

Reply:
If you want to use a regex for version numbers, there is a version number regex inside `rubygems` that it uses for the Gem::Version class. That is what I would suggest using 👍

Reply:
Refactored to use `Gem::Version::ANCHORED_VERSION_PATTERN`.